tests/integration/backup/test_v3_backup.py

```diff
@@ -334,6 +334,48 @@ def test_errors_when_omitting_replication_factor(config_file: Path) -> None:
     assert "the following arguments are required: --replication-factor" in exc_info.value.stderr.decode()
 
 
+def test_error_data(
+    config_file: Path,
+    admin_client: KafkaAdminClient,
+) -> None:
+    topic_name = "a-topic"
+    backup_directory = (
+        Path(__file__).parent.parent.resolve()
+        / "test_data"
+        / "backup_v3_corrupt_last_record_bit_flipped_no_checkpoints"
+        / topic_name
+    )
+    metadata_path = backup_directory / f"{topic_name}.metadata"
+
+    # Make sure topic doesn't exist beforehand.
+    try:
+        admin_client.delete_topics([topic_name])
+    except UnknownTopicOrPartitionError:
+        print("No previously existing topic.")
+    else:
+        print("Deleted topic from previous run.")
+
+    with patch("kafka.producer.kafka.KafkaProducer.send") as mock_send:
+        mock_send.side_effect = KafkaError()
+        with pytest.raises(subprocess.CalledProcessError) as er:
+            subprocess.run(
+                [
+                    "karapace_schema_backup",
+                    "restore",
+                    "--config",
+                    str(config_file),
+                    "--topic",
+                    topic_name,
+                    "--location",
+                    str(metadata_path),
+                ],
+                capture_output=True,
+                check=True,
+            )
+            print(er.value.stderr)
+            assert er.value.returncode == 3
```

Comment:
```suggestion
        print(er.value.stderr)
        assert er.value.returncode == 3
```

issue: This lines will never be reached, they must be outside the managed block.